lib/cloudregister/registerutils.py

```diff
@@ -173,6 +175,96 @@ def clear_rmt_as_scc_proxy_flag():
         pass
 
 
+# ----------------------------------------------------------------------------
+def clean_payg_extensions():
+    """Remove non free extensions files from /etc/products.d/."""
+    extensions = get_extensions()
+    products_paths = glob.glob('/etc/products.d/*.prod')
+
+    for extension in extensions:
+        # The following not free condition checks on
+        # the access capability of the extension
+        # This is a server side capability
+        if not extension.get('free', True):
+            extension_path = os.path.join(
+                os.sep, 'etc', 'products.d', '{}.prod'.format(
+                    extension.get('identifier')
+                )
+            )
+            if extension_path in products_paths:
+                os.unlink(extension_path)
+                logging.info(
+                    'Non free extension file %s removed' % extension_path
+                )
+
+
+# ----------------------------------------------------------------------------
+def get_product_tree(product_file=BASE_PRODUCT_PATH):
```

Comment:
What does moving the `get_product_tree` and `get_product_triplet` functions from the command implementation have to do with the cleanup of the non free extensions? This adds an unnecessary distraction to the task ad hand.

Reply:
> What does moving the get_product_tree and get_product_triplet functions from the command implementation have to do with the cleanup of the non free extensions?

`clean_payg_extensions` method calls `get_extensions` method which calls `get_product_data` method which calls `get_product_triplet`

Reply:
@rjschwei I'm about to close this thread as resolved now as I believe Jesus provided the needed information. As I said from my perspective the involved refactoring/code-moves are good for the requested change and finally also adds tests for crucial code like product triplet or SUSEConnect  invocation.

If this change should be split into a refactor first, let's discuss at the workshop. I would be good as I think the code move plus tests is still good to review